modules/autotagging-commons/src/main/java/org/opensearch/rule/action/TransportUpdateRuleAction.java

```diff
@@ -0,0 +1,106 @@
+     * @param rulePersistenceServiceRegistry - a {@link RulePersistenceServiceRegistry} object
+     * @param ruleRoutingServiceRegistry - a {@link RuleRoutingServiceRegistry} object
+     */
+    @Inject
+    public TransportUpdateRuleAction(
+        TransportService transportService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        RulePersistenceServiceRegistry rulePersistenceServiceRegistry,
+        RuleRoutingServiceRegistry ruleRoutingServiceRegistry
+    ) {
+        super(UpdateRuleAction.NAME, actionFilters, transportService.getTaskManager());
+        this.rulePersistenceServiceRegistry = rulePersistenceServiceRegistry;
+        this.ruleRoutingServiceRegistry = ruleRoutingServiceRegistry;
+        this.threadPool = threadPool;
+
+        transportService.registerRequestHandler(
+            UpdateRuleAction.NAME,
+            ThreadPool.Names.SAME,
+            UpdateRuleRequest::new,
+            new TransportRequestHandler<UpdateRuleRequest>() {
+                @Override
+                public void messageReceived(UpdateRuleRequest request, TransportChannel channel, Task task) {
+                    executeLocally(request, ActionListener.wrap(response -> {
+                        try {
+                            channel.sendResponse(response);
+                        } catch (IOException e) {
+                            logger.error("Failed to send UpdateRuleResponse to transport channel", e);
+                            throw new TransportException("Fail to send", e);
+                        }
+                    }, exception -> {
+                        try {
+                            channel.sendResponse(exception);
+                        } catch (IOException e) {
+                            logger.error("Failed to send exception response to transport channel", e);
+                            throw new TransportException("Fail to send", e);
+                        }
+                    }));
+                }
+            }
+        );
+    }
+
+    @Override
+    protected void doExecute(Task task, UpdateRuleRequest request, ActionListener<UpdateRuleResponse> listener) {
+        ruleRoutingServiceRegistry.getRuleRoutingService(request.getFeatureType()).handleUpdateRuleRequest(request, listener);
+    }
+
+    /**
+     * Executes the update rule operation locally on the dedicated rule thread pool.
+     * @param request the UpdateRuleRequest
+     * @param listener listener to handle response or failure
+     */
+    private void executeLocally(UpdateRuleRequest request, ActionListener<UpdateRuleResponse> listener) {
+        threadPool.executor(RULE_THREAD_POOL_NAME).execute(() -> {
+            final RulePersistenceService rulePersistenceService = rulePersistenceServiceRegistry.getRulePersistenceService(
+                request.getFeatureType()
+            );
+            rulePersistenceService.updateRule(request, listener);
+        });
+    }
```

Comment:
If we have two concurrent update request for the entity does it make sense to make this concurrent safe ?